### PR TITLE
discrete accuracy utilities

### PR DESCRIPTION
**Pull request by Shoeboxam**
Closes https://github.com/opendp/opendp/issues/577

### CODE REVIEW

File: rust/src/accuracy/mod.rs [hunk 1]
```diff
@@ -110,6 +268,20 @@ pub mod test {
...
+    fn test_comparison() -> Fallible<()> {
+        let alpha = 0.05;
+
+        let scale = 20.;
+        println!("cont accuracy: {}", laplacian_scale_to_accuracy(scale, alpha)?);
+        println!("disc accuracy: {}", discrete_laplacian_scale_to_accuracy(scale, alpha)?);
+
+        let accuracy = 20.;
+        println!("cont scale: {}", accuracy_to_laplacian_scale(accuracy, alpha)?);
+        println!("disc scale: {}", accuracy_to_discrete_laplacian_scale(accuracy, alpha)?);
```
Comment (andrewvyrros) [inline rust/src/accuracy/mod.rs:281]: How about gaussian?

Can we make any assertions of discrete vs continuous?

File: rust/src/accuracy/mod.rs [hunk 2]
```diff
@@ -237,8 +409,8 @@ pub mod test {
         let accuracy = 1.0;
         let theoretical_alpha = 0.05;
         let scale = accuracy_to_laplacian_scale(accuracy, theoretical_alpha)?;
-        let base_laplace = make_base_laplace::<AllDomain<f64>>(scale, None)?;
-        let n = 50_000;
+        let base_laplace = make_base_laplace::<AllDomain<f64>>(scale, Some(-100))?;
+        let n = 500_000;
```
Comment (andrewvyrros) [inline rust/src/accuracy/mod.rs:413]: These are a little slow at 500K, do you want this so big?

Similar for `test_empirical_discrete_gaussian_accuracy`.

**Review by andrewvyrros — COMMENTED** [review]
Code looks good, but we may want to have someone else look at the proofs.

File: rust/src/accuracy/accuracy_to_discrete_gaussian_scale.tex [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+the relationship between $\alpha$, $a$ and $scale$, is:
+
+\begin{align*}
+    1 - \alpha = \frac{\sum_{y=0}^{a - 1} (1 + 1[y = 0]) e^{-(y/s)^2/2}}{\sum_{z \in \mathbb{Z}} e^{-(z/s)^2/2}}
+\end{align*}
+
+A closed-form expression for s doesn't exist, 
+so we use a numerical approach by a binary search.
+An upper bound is provided by \rustdoc{accuracy/fn}{accuracy\_to\_laplacian\_scale}.
+The binary search finds the smallest $s$ such that
```
[the remark this answers is not in the document]
Comment (Shoeboxam) [reply rust/src/accuracy/accuracy_to_discrete_gaussian_scale.tex:22]: Fixed that link to refer to the gaussian. 

No dumb questions. I guess there is no "the" upper bound, as there can always be more than one, but there is only one tightest upper bound. But in this case, it referring specifically to "the" upper bound used in the binary search.

**Review by andrewvyrros — DISMISSED** [review]

**Review by andrewvyrros — APPROVED** [review]

**Pull request merged**